File: (03-25-2024)Merge Intervals/joshlin/solution.py
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals, key=lambda interval: interval[0])
        res = [intervals[0]]
        for interval in intervals[1:]:
            if res[-1][1] >= interval[0]:
                res[-1][1] = max(res[-1][1], interval[1])
            else:
                res.append(interval)
        
        return res

File: (03-25-2024)Merge Intervals/joshlin/solution.ts
function merge(intervals: number[][]): number[][] {
    intervals = intervals.map<number []>(a => [...a]).sort((a, b) => a[0] - b[0]);
    const res: number[][] = [];
    for (let i = 0; i < intervals.length; i++) {
        if (res.length != 0 && res[res.length-1][1] >= intervals[i][0]) {
            res[res.length-1][1] = Math.max(res[res.length-1][1], intervals[i][1]);
        } else {
            res.push(intervals[i]);
        };
    };
    return res;
};